### PR TITLE
Added Cart-Page with its Functionality

### PR DESCRIPTION
**Pull request by anshtripathi01**
**Please Review**

### CODE REVIEW

File: src/CartUtilityFunction.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    fetch("/api/user/cart", {
+        method: "POST",
+        body: JSON.stringify({ product: item }),
+        headers: {
+            authorization: encodedToken,
+            "Content-type": "application/json; charset=UTF-8"
+        }
+    })
+        .then(res => res.json())
+        .then(data => console.log(data))
```
Comment (shashikantmaurya0007) [inline src/CartUtilityFunction.js:14]: be consistent use axios or fetch throughout the application

File: src/pages/Cart/Cart.css [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+.count {
+    padding: 0.2rem 1rem;
+}
+
+.counter-btn {
+    width: 1.6rem;
+    height: 1.6rem;
+    border-radius: 50%;
+    border: 1px solid #808080;
+    cursor: pointer;
```
Comment (agrawalshanu708) [inline src/pages/Cart/Cart.css:47]: use variable color here

**Review by agrawalshanu708 — APPROVED** [review]
format your code
you can merge it

File: src/Components/ProductCard.js [hunk 3]
```diff
@@ -1,8 +1,54 @@
...
+  const { state , dispatch} = useContext(StateContext)
+  const {token,setToken} = useContext(AuthContext);
+  window.onbeforeunload = function (e) {
+    localStorage.clear()
+    setToken(undefined);
+
+};
+
+  
+  let navigate = useNavigate();
```
Comment (shashikantmaurya0007) [inline src/Components/ProductCard.js:20]: use const here

**Review by shashikantmaurya0007 — APPROVED** [review]
added few comments for improvement rest looks good

File: src/CartUtilityFunction.js (hunk 1 above)
Comment (anshtripathi01) [reply src/CartUtilityFunction.js:14]: ok

Comment (anshtripathi01) [reply src/CartUtilityFunction.js:14]: thank you

File: src/Components/ProductCard.js (hunk 3 above)
Comment (anshtripathi01) [reply src/Components/ProductCard.js:20]: fixed

**Pull request merged**